common/types/hashes.go

```diff
@@ -177,6 +242,12 @@ func (h *Hash32) SetBytes(b []byte) {
 	copy(h[32-len(b):], b)
 }
 
+func (h Hash32) ToHash20() Hash20 {
```

Comment:
maybe we could have copyyohash20 and tohash20 where the latter doesn't copy but use the same array. of course only if we have a use for it.

Reply:
this is the safest way to do it and not risk leaking bytes IMO. if we need better performance we could always change it to be non copy

Reply:
We should prefer copy over referencing such data.
As I have said before, Hash* should behave like an int (copy) wherever possible.